src/Compiler/CompilerOptions.cpp

```diff
@@ -177,6 +178,16 @@ static llvm::cl::opt<bool, true> enableONNXHybridPassOpt("onnx-hybrid-pass",
     llvm::cl::location(enableONNXHybridPass), llvm::cl::init(true),
     llvm::cl::cat(OnnxMlirCommonOptions));
 
+static llvm::cl::list<std::string, std::vector<std::string>>
+    onnxHybridPassDisableClassesOpt("onnx-hybrid-pass-disable-class",
+        llvm::cl::desc("Class of hybrid pass patterns to disable.\n"
+                       "Can be shape-inference, canonicalization, "
+                       "constant-propagation, decomposition.\n"
+                       "Repeat the flag to disable multiple classes."),
+        llvm::cl::value_desc("class of hybrid pass patterns to disable"),
+        llvm::cl::location(onnxHybridPassDisableClasses),
+        llvm::cl::cat(OnnxMlirCommonOptions));
```

Comment:
I would make this option a sub-option of `-onnx-hybrid-transform` instead of a parallel option here, e.g. `--onnx-hybrid-transform=disable-class=constant-propagation,canonicalization`

Reply:
ok

done

Reply:
you can call it with `--onnx-hybrid-transform="constant-propagation=false canonicalization=false"`

Reply:
Thanks!